authapi/tests/test_api.py

```diff
@@ -1001,3 +1002,62 @@ def test_remove_user_from_other_organizations_team(self):
                 'seedorganization-teams-users-detail',
                 args=[org2.pk, team.pk, user.pk]))
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class TokenTests(AuthAPITestCase):
+    def test_create_token(self):
+        '''Sending a valid email and password should create a token for that
+        user.'''
+        data = {
+            'email': 'test@example.org',
+            'password': 'testpass',
+        }
+        user = User.objects.create_user(
+            username=data['email'], email=data['email'],
+            password=data['password'])
+
+        response = self.client.post(reverse('create-token'), data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        [token] = Token.objects.filter(user=user)
+        self.assertEqual(token.key, response.data['token'])
+
+    def test_create_token_invalid_user(self):
+        '''An invalid email, or incorrect password should return an
+        unauthorized response.'''
+        response = self.client.post(reverse('create-token'), data={
+            'email': 'foo@bar.com', 'password': 'foo'})
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create_token_inactive_user(self):
+        '''A user that is not active, should get a forbidden response to
+        creating a token.'''
+        data = {
+            'email': 'test@example.org',
+            'password': 'testpass',
+        }
+        User.objects.create_user(
+            username=data['email'], email=data['email'],
+            password=data['password'], is_active=False)
+
+        response = self.client.post(reverse('create-token'), data=data)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_token_removes_other_tokens(self):
+        '''When a new token for a user is requested, all other tokens for
+        that user should be removed.'''
+        data = {
+            'email': 'test@example.org',
+            'password': 'testpass',
+        }
+        user = User.objects.create_user(
+            username=data['email'], email=data['email'],
+            password=data['password'])
+        first_token = Token.objects.create(user=user)
+
+        response = self.client.post(reverse('create-token'), data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        [token] = Token.objects.filter(user=user)
+        self.assertEqual(token.key, response.data['token'])
+        self.assertNotEqual(first_token.key, token.key)
```

Comment:
I feel like `POST` is a better choice, since you're creating a new token, not updating a token (at least for the first time that you create the token). I like `/user/token/`, I think I'll change it to that.